packages/ember-metal/lib/meta.js

```diff
@@ -326,7 +327,7 @@ export class Meta {
     let descriptor = lookupDescriptor(obj, key);
     let isMandatorySetter = descriptor && descriptor.set && descriptor.set.isMandatorySetter;
 
-    if (isMandatorySetter) {
+    if (isMandatorySetter === true) {
```

Comment:
For `r && l`, if both `r` and `l` are known to be Boolean (i.e. because they are `!==` comparisons), then the compiler knows that the result `r && l` will also be a Boolean.